### PR TITLE
Dedupe winapi

### PR DESCRIPTION
**Pull request by Eijebong**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23449)

### CODE REVIEW

**Comment by Eijebong** [comment]
Tidy failed because winapi isn't duped anymore :laughing:

File: Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,5 @@ opt-level = 3
 #
 #     [patch."https://github.com/servo/<repository>"]
 #     <crate> = { path = "/path/to/local/checkout" }
+mio = { git = "https://github.com/servo/mio.git", branch = "servo" }
+iovec = { git = "https://github.com/servo/iovec.git", branch = "servo" }
```
Comment (SimonSapin) [inline Cargo.toml]: Forking two repositories is unfortunate. What’s the plan to unfork? Are there PRs waiting to be merged upstream?

Comment (Eijebong) [reply Cargo.toml]: As soon as mio get a new release and is updated in tokio we can unfork those. 

The master branches of mio and iovec already contain the patches but also other breaking changes so we can't use them directly. Mio had promised a new release more than one year ago so hopefully it'll come soon...

The duplication of was blocking @paulrouget on some work hence the forking

Comment (SimonSapin) [reply Cargo.toml]: Sounds reasonable, thanks. Could you add a comment in `Cargo.toml` that explains this winapi situation? r=me with that.

Comment (Eijebong) [reply Cargo.toml]: Is it enough now ?

**Comment by jdm** [comment]
@bors-servo treeclosed=9000
Doing some upgrades.

**Comment by jdm** [comment]
@bors-servo retry
* network issues

**Pull request merged**